File: phase-2-persistence/2.3-round-trip-tests/starter/Kingdom.Engine/Buildings/Farm.cs
namespace Kingdom.Engine.Buildings;

public class Farm : Building
{
    public Farm(string name) : base(name) { }
    public Farm(string name, int level) : base(name, level) { }

    public override void Tick(ResourceLedger ledger)
    {
        ledger.Add(Resource.Food, 5 * Level);
    }
}

File: phase-5-roblox/5.3-oop-in-luau/starter/roblox-kingdom/Engine/Farm.lua
local Building = require(script.Parent.Building)

local Farm = setmetatable({}, { __index = Building })
Farm.__index = Farm

function Farm.new(name: string)
    local self = Building.new(name)
    setmetatable(self, Farm)
    return self
end

function Farm:tick(ledger: any)
    ledger:add("Food", 5 * self.level)
end

return Farm
